backtrace-library/src/main/java/backtraceio/library/metrics/UniqueEvent.java

```diff
@@ -0,0 +1,56 @@
+package backtraceio.library.metrics;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import backtraceio.library.base.BacktraceBase;
+import backtraceio.library.common.BacktraceStringHelper;
+import backtraceio.library.logger.BacktraceLogger;
+
+public class UniqueEvent extends Event {
+
+    private final static transient String LOG_TAG = UniqueEvent.class.getSimpleName();
+
+    /**
+     * Unique events API spec requires unique events to be a JSON array, but we still treat it as a single string
+     */
+    @SerializedName("unique")
+    private final List<String> name;
```

Comment:
I thought we want to store a single name for a unique event.

Reply:
Discussed offline, the spec requires unique events names to be a JSON array so we keep it this way for GSON serializer. I will try to find ways to clean this up @konraddysput but I don't think we can avoid keeping this class member as an array if we want to use GSON serializer.